ray-curator/ray_curator/stages/deduplication/removal.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Removal stage for distributed deduplication pipeline.
+
+This stage implements the removal phase of the distributed deduplication approach:
+1. Takes a DocumentBatch and determines the min/max ID range
+2. Filters the parquet files for IDs to remove within this range
+3. Filters out documents based on the removal list
+4. Returns the filtered DocumentBatch
+"""
+
+import time
+from dataclasses import dataclass
+from typing import Any
+
+import pandas as pd
+
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.deduplication.id_generator import CURATOR_DEDUP_ID_STR
+from ray_curator.tasks import DocumentBatch
+
+
+@dataclass
+class DuplicatesRemovalStage(ProcessingStage[DocumentBatch, DocumentBatch]):
```

Comment:
@sarahyurick / @huvunvidia i'll move this to `stages/text/deduplication/removal.py` and call it TextDuplicatesRemovalStage, and in Image you can follow suite.